### PR TITLE
Update/0.18.0

### PR DESCRIPTION
**Pull request by Dominion5254**
- Update LND to 0.18.0
- Update buildService and releaseService actions
- Add missing migration
- Fix build errors

### CODE REVIEW

File: manifest.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 id: lnd
 title: LND
-version: 0.17.5.1
+version: 0.18.0
 release-notes: |-
+  * Update to 0.18.0 [Release Notes](https://github.com/lightningnetwork/lnd/releases/tag/v0.18.0-beta)
   * Add Umbrel 1.0 migration action
```
Comment (ProofOfKeags) [inline manifest.yaml:6]: shouldn't this be removed?

File: .github/workflows/buildService.yml [hunk 2]
```diff
@@ -19,12 +19,6 @@ jobs:
       - name: Checkout services repository
         uses: actions/checkout@v4
 
-      - name: Debug
-        run: |
-          pwd
-          ls -alR
-
-
```
Comment (ProofOfKeags) [inline .github/workflows/buildService.yml:27]: 🫡

File: .github/workflows/buildService.yml [hunk 3]
```diff
@@ -33,9 +27,11 @@ jobs:
...
           echo "package_id=$PACKAGE_ID" >> $GITHUB_ENV
+          printf "\n SHA256SUM: $(sha256sum ${PACKAGE_ID}.s9pk) \n"
         shell: bash
+
       - name: Upload .s9pk
         uses: actions/upload-artifact@v4
         with:
           name: ${{ env.package_id }}.s9pk
-          path: ./${{ env.package_id }}.s9pk
+          path: ./${{ env.package_id }}.s9pk
```
Comment (ProofOfKeags) [inline .github/workflows/buildService.yml:37]: unintentional?

File: .github/workflows/releaseService.yml [hunk 4]
```diff
@@ -68,4 +69,4 @@ jobs:
             echo "Publish skipped: missing registry credentials."
           else
             start-sdk publish https://$S9USER:$S9PASS@$S9REGISTRY ${{ env.package_id }}.s9pk
-          fi
+          fi
```
Comment (ProofOfKeags) [inline .github/workflows/releaseService.yml:72]: seems like maybe your autoformatter might be conflicting with someone else's?

File: configurator/src/main.rs [hunk 5]
```diff
@@ -406,7 +406,7 @@ fn main() -> Result<(), anyhow::Error> {
         alias = alias,
         color = config.color,
         feeurl_row = if use_neutrino {
-            "feeurl=https://nodes.lightning.computer/fees/v1/btc-fee-estimates.json"
+            "fee.url=https://nodes.lightning.computer/fees/v1/btc-fee-estimates.json"
```
Comment (ProofOfKeags) [inline configurator/src/main.rs:409]: 🫡

File: scripts/services/migrations.ts [hunk 6]
```diff
@@ -210,6 +210,33 @@ export const migration: T.ExpectedExports.migration = compat.migrations
           throw new Error("Cannot downgrade");
         },
       },
+      "0.18.0": {
+        up: compat.migrations.updateConfig(
+          (config: any) => {
+            config.advanced["protocol-zero-conf"] = false;
+            config.advanced["protocol-option-scid-alias"] = false;
```
Comment (ProofOfKeags) [inline scripts/services/migrations.ts:217]: Worth noting that zero-conf requires option-scid-alias. I know migrations isn't the right place to handle that, but probably needs to be added in the config validation scripts.

The check would be something like `!protocol-zero-conf || protocol-option-scid-alias` which would need to resolve to true for the config to validate properly.

**Review by ProofOfKeags — COMMENTED** [review]
Small UX note regarding ensuring that if zero-conf is enabled you must also have scid-alias enabled as well.

EDIT: Also to take full advantage of the new update I recommend adding the [`sweeper`](https://github.com/lightningnetwork/lnd/blob/master/sample-lnd.conf#L1674-L1729) options into the advanced section of the config.

File: manifest.yaml (hunk 1 above)
Comment (Dominion5254) [reply manifest.yaml:6]: A note about upstream updates is usually included in the service release notes alongside a link to the upstream release notes - why would we not include that note here?

Or are you suggesting the note be something more explicit like `Update LND to 0.18.0`?

File: .github/workflows/buildService.yml (hunk 3 above)
Comment (Dominion5254) [reply .github/workflows/buildService.yml:37]: This like the deletion of the trailing whitespace flagged. This should be fine though.

File: .github/workflows/releaseService.yml (hunk 4 above)
Comment (Dominion5254) [reply .github/workflows/releaseService.yml:72]: Same as above, it looks like the deletion of the trailing whitespace flagged.

File: scripts/services/migrations.ts (hunk 6 above)
Comment (Dominion5254) [reply scripts/services/migrations.ts:217]: We have a validation for `protocol-opiton-scit-alias` and `protocol-no-anchors` in [setConfig.ts](https://github.com/Start9Labs/lnd-startos/blob/update/0.18.0/scripts/services/setConfig.ts#L34-L47).

Toast errors aren't really optimal, but I believe this is the best that can be done for dynamic config requirements until 0.3.6.

getConfig.ts also notes the requirement in the description [here](https://github.com/Start9Labs/lnd-startos/blob/update/0.18.0/scripts/services/setConfig.ts#L34-L47)

**Comment by Dominion5254** [comment]
If I understand correctly, exposing the `sweeper` config options basically allow users to potentially optimize/save on sweep fees with the tradeoff of potential loss of funds if the values are set too low. Am I understanding this correctly, or is that only a concern for `sweeper.maxfeerate`?

I would be somewhat hesitant to expose users that potential footgun and the resulting support load.

File: scripts/services/migrations.ts (hunk 6 above)
Comment (ProofOfKeags) [reply scripts/services/migrations.ts:217]: Cool yeah this covers my issue.

**Comment by ProofOfKeags** [comment]
> If I understand correctly, exposing the sweeper config options basically allow users to potentially optimize/save on sweep fees with the tradeoff of potential loss of funds if the values are set too low. Am I understanding this correctly, or is that only a concern for sweeper.maxfeerate?

These values help inform the sweeper how to make decisions wrt how much it burns in chain fees in order to recover possibly contested outputs (HTLCs and Breach outputs). Unfortunately there's no "correct" value, which is precisely why we expose it. There is risk of fund loss for all values via two vectors: unclaimed outputs, and overpaying for fees.

 > I would be somewhat hesitant to expose users that potential footgun and the resulting support load.

This is fair. I also would like to see every *user* option exposed eventually. I figure the "Advanced" section of the menu should be where this stuff lives anyway. What kind of assurance are you looking for here?

File: manifest.yaml (hunk 1 above)
Comment (remcoros) [reply manifest.yaml:6]: I think he means "Add Umbrel 1.0 migration action"

Comment (ProofOfKeags) [reply manifest.yaml:6]: Correct, this is in reference to the Umbrel action.

Comment (Dominion5254) [reply manifest.yaml:6]: Ah. That is correct as is.

The Umbrel 1.0 migration was merged into main in a previous [PR](https://github.com/Start9Labs/lnd-startos/pull/124), but 0.18.0 will be the first release including this action.

File: configurator/src/lnd.conf.template [hunk 7]
```diff
@@ -74,6 +74,14 @@ protocol.option-scid-alias={protocol_option_scid_alias}
 protocol.zero-conf={protocol_zero_conf}
 protocol.simple-taproot-chans={protocol_simple_taproot_chans}
 
+[sweeeper]
```
Comment (ProofOfKeags) [inline configurator/src/lnd.conf.template:77]:
```suggestion
[sweeper]
```

File: configurator/src/main.rs [hunk 8]
```diff
@@ -131,6 +131,17 @@ struct BitcoinChannelConfig {
...
+#[derive(serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct SweeperConfig {
+    sweeper_maxfeerate: u64,
+    sweeper_nodeadlineconftarget: usize,
+    sweeper_budget_tolocalratio: f64,
+    sweeper_budget_anchorcpfpratio: f64,
+    sweeper_budget_deadlinehtlcratio: f64,
+    sweeper_budget_nodeadlinehtlcratio: f64,
+}
```
Comment (ProofOfKeags) [inline configurator/src/main.rs:143]: I think you also want to include `Option<usize>` values for the different fee caps

**Pull request merged**